### PR TITLE
Enhancement/rendering2 working version

### PR DESCRIPTION
**Pull request by tieswesterborg**
Rendering enhancement.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -30,4 +30,5 @@ hs_err_pid*
 .classpath
 .project
 .settings/
-*.gfa
+TB328.gfa
+src/main/resources/TB328.gfa
```
Comment (ghost) [inline .gitignore:34]: Why two ignores for the TB328?

File: src/main/java/application/controllers/MainController.java [hunk 2]
```diff
@@ -474,8 +474,16 @@ private void setAnnotationButtonsActionListener(Button highlightButton, Button d
 
                     // Deselect the previously highlighted annotation as only one should be highlighted at a time.
                     deselectAllAnnotations();
+
+                    if(newAnnotation.getSpannedNodes().get(0) != null) {
```
Comment (ghost) [inline src/main/java/application/controllers/MainController.java:478]: Space needed after the if

File: src/main/java/application/fxobjects/Edge.java [hunk 3]
```diff
@@ -123,4 +123,17 @@ public void setLine(Line line) {
...
     }
+
+    /**
+     * Returns the length of this Edge.
+     *
+     * @return the Edge.
+     */
+    public double getLength() {
+        double xLength = getLine().endXProperty().get()
+                - getLine().startXProperty().get();
```
Comment (ghost) [inline src/main/java/application/fxobjects/Edge.java:134]: Can be placed on one line

File: src/main/java/application/fxobjects/Edge.java [hunk 4]
```diff
@@ -123,4 +123,17 @@ public void setLine(Line line) {
...
+        double yLength = getLine().endYProperty().get()
+                - getLine().startYProperty().get();
```
Comment (ghost) [inline src/main/java/application/fxobjects/Edge.java:136]: Can be placed on one line

File: src/main/java/core/graph/Graph.java [hunk 5]
```diff
@@ -108,6 +112,7 @@ public Boolean addGraphComponents(ArrayList<String> ref, int depth) {
                 //LoadOneUp is only needed when we do not start on the top level.
                 loadBoth(depth);
             } else { //Second time. All models are loaded
+
```
Comment (ghost) [inline src/main/java/core/graph/Graph.java]: New line can be removed

File: src/main/java/core/graph/Graph.java [hunk 6]
```diff
@@ -200,6 +205,7 @@ public Model generateModel(ArrayList<String> ref, int depth, Model toret) {
         //Select the level to draw from
         HashMap<Integer, Node> nodeMap = levelMaps.get(depth);
 
+
```
Comment (ghost) [inline src/main/java/core/graph/Graph.java]: New line can be removed

File: src/main/java/core/graph/GraphReducer.java [hunk 7]
```diff
@@ -60,18 +60,28 @@ public static void determineParents(HashMap<Integer, Node> nodeMap) {
...
         levelMaps.add(startMap);
         startMapSize = startMap.size();
+        int reduceAmount = 5;
 
         for (int i = 1;; i++) {
             HashMap<Integer, Node> levelMap = collapse(levelMaps.get(i - 1), i - 1);
             int previousMapSize = levelMaps.get(i - 1).size();
             int currentMapSize = levelMap.size();
 
+            if (levelMaps.size() == 25) {
```
Comment (ghost) [inline src/main/java/core/graph/GraphReducer.java:70]: Consider making a constance for the max number of level maps as the value is also duplicate on line 81

File: src/main/java/core/graph/GraphReducer.java [hunk 8]
```diff
@@ -60,18 +60,28 @@ public static void determineParents(HashMap<Integer, Node> nodeMap) {
...
+                i -= reduceAmount;
+            }
+
             // Don't make any new zoom level if the number of nodes after reduction is only 2 less
             // than the number of nodes after previous reduction.
             if ((previousMapSize - currentMapSize) <= minDelta) {
                 levelMaps.set(i - 1, levelMap);
                 int maxDepth = 20;
                 for (int j = i; maxDepth < 1001; j++) {
+                    if (levelMaps.size() == 25) {
```
Comment (ghost) [inline src/main/java/core/graph/GraphReducer.java:81]: See comment above

File: src/main/java/core/model/Model.java [hunk 9]
```diff
@@ -392,4 +406,35 @@ public Cell getLeftMost() {
...
+
+    /**
+     * Method to add a cell to a Tile
+     *
+     * @param tile to the to add to.
+     * @param c the Cell to add.
+     */
+    public void addCellInTile(int tile, Cell c) {
+        tileCellMap.putIfAbsent(tile, new ArrayList<>());
+
```
Comment (ghost) [inline src/main/java/core/model/Model.java]: new line can be removed

File: src/main/java/application/controllers/GraphController.java [hunk 10]
```diff
@@ -316,7 +316,7 @@ public void update(ArrayList<String> ref, int depth) {
         } else {
             //if(hValue != lastDrawnHValue) {
```
Comment (ghost) [inline src/main/java/application/controllers/GraphController.java:315]: Commented code has to be removed

File: src/main/java/application/controllers/GraphController.java [hunk 11]
```diff
@@ -357,19 +357,19 @@ private void checkEdgeLength(Edge e, double maxEdgeLength, double maxEdgeLengthL
...
-            root.getChildren().addAll(graph.getModelAllInView(min, max).getAddedCells());
-        } else {
-            root.getChildren().addAll(graph.getModelAddedInView(min, max).getAddedEdges());
-            root.getChildren().addAll(graph.getModelAddedInView(min, max).getAddedCells());
-        }
+        //if (graph.getModel().getAllCells().size() > 0) {
+            Model model = graph.getModelAllInView(min);
+            root.getChildren().addAll(model.getAddedEdges());
+            root.getChildren().addAll(model.getAddedCells());
+//        } else {
```
Comment (ghost) [inline src/main/java/application/controllers/GraphController.java]: Commented code has to be removed

File: src/test/java/core/graph/GraphTest.java [hunk 12]
```diff
@@ -220,25 +220,15 @@ public void testGetMaxWidth() {
...
-    /**
-     * Tests the getModelAllInView method.
-     */
-    @Test
-    public void testGetModelAllInView() {
-        Model m = g.getModel();
-        Model res = g.getModelAllInView(0, Integer.MAX_VALUE);
-        assertNotEquals(m, res);
-    }
+//    /**
```
Comment (ghost) [inline src/test/java/core/graph/GraphTest.java:223]: More commented code, should be removed

File: src/main/java/application/controllers/GraphController.java (hunk 10 above)
Comment (tieswesterborg) [inline src/main/java/application/controllers/GraphController.java:315]: See issue 159

File: src/test/java/core/graph/GraphTest.java (hunk 12 above)
Comment (tieswesterborg) [inline src/test/java/core/graph/GraphTest.java:223]: This test should be edited, not deleted. This can be done in a later PR

**Comment by ghost** [comment]
Fix OK

**Pull request merged**